lib/ansiblelint/versiontools.py

```diff
@@ -0,0 +1,37 @@
+"""Version tools set."""
+
+import contextlib
+import os
+
+from setuptools_scm import get_version
+
+
+def get_version_from_scm_tag(
+        root='.',
+        relative_to=None,
+        local_scheme='node-and-date',
+):
+    """Retrieve the version from SCM tag in Git or Hg."""
+    with contextlib.suppress(LookupError):
+        return get_version(
+            root=root,
+            relative_to=relative_to,
+            local_scheme=local_scheme,
+        )
+
+    return 'unknown'
+
+
+def cut_local_version_on_upload(version):
```

Comment:
I see this boilerplate added again and again in multiple places. What prevents adding it directly to setuptools_scm so we don't need it in each project?

Reply:
Feel free to open an issue there. I'm working with what I've got.